src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3075,7 +3082,14 @@ def unlock(self, job, id, options):
 
         if options['key_file']:
             keys_supplied = self._retrieve_keys_from_compressed_file(job, id)
-        keys_supplied.update({d['name']: d['passphrase'] for d in options['datasets']})
+
+        for i, ds in enumerate(options['datasets']):
+            if all(ds.get(k) for k in ('key', 'passphrase')):
+                verrors.add(
+                    f'unlock_options.datasets.{i}.dataset.key',
+                    f'Must not be specified when unlock_options.datasets.{i}.dataset.passphrase is supplied.'
```

Comment:
We could name it as "Must not be specified when passphrase for {dataset} is supplied"

unlock_options.datasets.{i}.dataset.passphrase as part of the text that would eventually show up in the UI is weird.